### DISCUSSION TITLE
some fixes to gdk-pixbuf2 and graphviz

### DISCUSSION THREAD

**Discussion by revelator** [msys2/MINGW-packages#20791] [Ideas]
Finally nailed why gdk-pixbuf2 fails to load plugins when linked to the static runtimes.
The gobject loader looks for a dllmain in the plugin but several plugins do not have one when linked with the static runtimes -> eg. -static-libgcc etc. this is due to many of the image libraries like tiff not having one as well so gdk-pixbuf2 tries to fool these by wrapping a DllMain on top of the libraries main function. for some reason which is not clear to me this works for 64 bit builds and not 32 bit.

so i did some work on the DllMain patch adding an entrypoint to DllMain and also duplicating the mechanism in the query loaders executable. it has worked beautifully since so i would highly recommend using the updated patch.
[0003-fix-dllmain.zip](https://github.com/msys2/MINGW-packages/files/15195251/0003-fix-dllmain.zip)

it can be dropped in as a replacement for the old patch just remember to update the pkgsums.

the graphviz fix was due to updated groff no longer defaulting to "-t" or array anymore so it fails to build the pdf documentation.

[0007-fix-groff-tables.zip](https://github.com/msys2/MINGW-packages/files/15195275/0007-fix-groff-tables.zip)

this should fix it.

**Comment by Biswa96** [comment]
Is there any issue with those mingw packages?

**Comment by revelator** [comment]
if linking with static libgcc 32 bit gdk-pixbuf2 fails to load several plugins so yeah.
graphviz cannot build the pdf documentation with groff as it is now (build error due to groff not defaulting to -t like older versions did).

**Reply by Biswa96** [reply]
mingw packages are not linked statically.
graphviz has been updated recently. Could you reproduce the issue with it? If yes please create an issue with proper steps to reproduce it and create a pull request with that patch.

**Comment by revelator** [comment]
even if they are not it is still a bug :) since it does not use the DllMain wrapper in gdk-pixbuf2 even in a normal build, it uses the DllMain from whatever library happens to have one and this gets visible when linked to the static runtime. (no it is not linked statically only libgcc is).

i found the bug in the most recent graphviz version 10.0.0.1 the groff tables are created like this ->` $(GROFF) -Tps -man $< > $$psfile || { rm -f $$psfile; exit 1; };` but our current groff needs this -> `$(GROFF) -Tps -t -man $< > $$psfile || { rm -f $$psfile; exit 1; };` notice the '-t'.

**Comment by revelator** [comment]
same in version 11.0.0 of graphviz so still needs the '-t' im afraid.

**Comment by revelator** [comment]
[loaders.zip](https://github.com/msys2/MINGW-packages/files/15202923/loaders.zip)
this is the loader.cache after the patch before it could not even load half of the image formats, the ones that failed were tiff webp avif heif svg jp2 jxl so quite a few. even if it works for you as it is this points to some problem with static runtimes when people want to avoid using shared libgcc for say there own projects. strangely it only seems to affect the 32 bit compiler and only with gnome software. qt5/6 is not affected and happily builds linked to both the static libgcc libstdc++ runtimes.
